klog.go

```diff
@@ -1577,7 +1578,7 @@ type KMetadata interface {
 
 // KObj returns ObjectRef from ObjectMeta
 func KObj(obj KMetadata) ObjectRef {
-	if obj == nil {
+	if val := reflect.ValueOf(obj); obj == nil || (val.Kind() == reflect.Ptr && val.IsNil()) {
```

Comment:
we should do the cheap check before making reflect calls

```
if obj == nil {
  return ObjectRef{}
}
if val := reflect.ValueOf(obj); (val.Kind() == reflect.Ptr || val.Kind() == reflect.Interface) && val.IsNil() {
  return ObjectRef{}
}
```

Reply:
done